Libplanet/Crypto/BlsSignatureExtensions.cs

```diff
@@ -0,0 +1,34 @@
+using System.Diagnostics.Contracts;
+using System.Linq;
+
+namespace Libplanet.Crypto
+{
+    /// <summary>
+    /// The extension class for <see cref="BlsSignature"/>. Mainly it defines the extensions for
+    /// aggregating signatures with conveniences.
+    /// <para>See <see cref="BlsSignature.AggregateVerify"/> or
+    /// <see cref="BlsSignature.FastAggregateVerify"/> for detailed usage of aggregated signature.
+    /// </para>
+    /// </summary>
+    public static class BlsSignatureExtensions
+    {
+        /// <summary>
+        /// Aggregates the given multiple signatures into one. An aggregated signature can be
+        /// validated with their pair public keys with <see cref="BlsSignature.AggregateVerify"/>
+        /// if the messages are unique, or <see cref="BlsSignature.FastAggregateVerify"/> if all the
+        /// messages are identical.
+        /// </summary>
+        /// <param name="signatures">The signatures to aggregate.</param>
+        /// <returns>Returns an aggregated signature of given signatures.</returns>
+        /// <remarks>Note that if any given signature is <b>infinite</b> value, then the
+        /// aggregated signature will be the aggregated signature without <b>infinite</b> signature.
+        /// This should be considered before aggregates the signature.</remarks>
+        [Pure]
+        public static BlsSignature AggregateAll(this BlsSignature[] signatures) =>
+            signatures.Aggregate((x, y) => x.AggregateSignature(y));
```

Comment:
This would depend on its usage, but there may be a better suited enumerable type instead of `[]`.

Reply:
https://github.com/planetarium/libplanet/blob/bcf03ace27e8621d2f8c2a4603e230d2928371c8/Libplanet/Crypto/BlsSignatureExtensions.cs#L10-L11

How about this?

Reply:
Depends on usage. Usually enforcing an immutable type is safer but with worse performance. I don't know enough about the exact context in which this will be used, but from the looks of it, this might get called on a "pool" of signatures that can change often. The `IEnumerable` signature signals that this method should be used synchronously or with some sort of locking mechanism from the caller.

Reply:
I think this should be okay. The `BlsSignature`s are converted to a native signature type, which is a separate object with `BlsSignature` and `BlsSignature` is only used for a 'referencing'.

https://github.com/planetarium/libplanet/blob/7080503a74570ae1774297eacf35d554d8896f65/Libplanet/Crypto/BlsCryptoBackend.cs#L117-L138

Reply:
- I think this can be used in asynchronous and synchronous in both situation. `BlsSignature` is immutable. The `AggregateSignature` internally creates new two native signatures from BLS library, and aggregates signatures, which is separate object from given `BlsSignature`s, and the results is packed into new `BlsSignature` and returns it.

- Does _`IEnumerable` is signaling the method should be used synchronously_ mean that LINQ does not support asynchronous operation?